File: app/src/main/java/com/jihoonyoon/soo/notepad/database/DietDao.java
package com.jihoonyoon.soo.notepad.database;

import com.jihoonyoon.soo.notepad.models.Diet;
import com.jihoonyoon.soo.notepad.models.Diet_Table;
import com.raizlabs.android.dbflow.sql.language.SQLite;

import java.util.Calendar;
import java.util.List;

public class DietDao {

    public static Diet getDietByCalendar(Calendar calendar){
        return SQLite.select().from(Diet.class).where(Diet_Table.calendar.is(calendar)).querySingle();
    }

    public static void removeDietByCalendar(Calendar calendar){
        SQLite.delete().from(Diet.class).where(Diet_Table.calendar.is(calendar)).execute();
    }

    public static List<Diet> getAllDiet(){
        return SQLite.select().from(Diet.class).queryList();
    }
}

File: app/src/main/java/com/jihoonyoon/soo/calendar/dao/DietDao.kt
//package com.jihoonyoon.soo.calendar.dao
//
//import androidx.room.*
//import com.jihoonyoon.soo.calendar.model.Diet
//import java.util.*
//
//@Dao
//interface DietDao {
//
//    @Insert
//    fun insertDiet(diet: Diet)
//
//    @Delete
//    fun deleteDiet(diet: Diet)
//
//    @Update
//    fun updateDiet(diet: Diet)
//
//    @Query("Select * From Diet")
//    fun selectDiet(): List<Diet>
//
//    @Query("Select * From Diet Where calendar = :calendar")
//    fun selectDietByCalendar(calendar: Calendar): Diet
//}